drivers/arm/i2c_master.c

```diff
@@ -71,27 +69,60 @@ __attribute__((weak)) void i2c_init(void) {
 }
 
 i2c_status_t i2c_start(uint8_t address) {
-    i2c_address = address;
+#if I2C_USE_MUTUAL_EXCLUSION
+    i2cAcquireBus(&I2C_DRIVER);
+#endif
+
     i2cStart(&I2C_DRIVER, &i2cconfig);
     return I2C_STATUS_SUCCESS;
 }
 
 i2c_status_t i2c_transmit(uint8_t address, const uint8_t* data, uint16_t length, uint16_t timeout) {
-    i2c_address = address;
+    msg_t        status;
```

Comment:
We dont target C89, any reason to move the declaration of the variables?

Reply:
None at all, was probably because I'd been writing C89 earlier in the day for work or something.
Will rollback those to minimise SLOC changes.